addons/mail/static/src/models/emoji.js

```diff
@@ -2,25 +2,28 @@
 
 import { registerModel } from '@mail/model/model_core';
 import { attr, many, one } from '@mail/model/model_field';
-import { replace } from '@mail/model/model_field_command';
+import { insert, replace } from '@mail/model/model_field_command';
 
 registerModel({
     name: 'Emoji',
     identifyingFields: ['unicode'],
     recordMethods: {
-        /**
-         * @private
-         * @returns {FieldCommand}
-         */
         _computeEmojiRegistry() {
             return replace(this.messaging.emojiRegistry);
         },
+        _computeEmojiCategories() {
+            return (insert({ categoryName: this.defaultEmojiCategory }));
+        },
     },
     fields: {
         description: attr({
             readonly: true,
         }),
+        defaultEmojiCategory: attr({
+            default: "all"
+        }),
         emojiCategories: many('EmojiCategory', {
+            compute: "_computeEmojiCategories",
             inverse: 'allEmojis',
```

Comment:
This is not correct.

1. `emojiCategories` relies on both `compute` and imperative changes, so it's wrong. When you define a computed field, you must make it complete (i.e. `insertAndReplace`, `clear`, or `replace`).
2. "all" category should not be a magic string from data received somewhere else. Either `emoji_data` should be adapted to have `all` category in all the emoji data, or don't use this `all` category in data and instead define proper field just for this category. That way, here `Emoji/emojiCategories` is computed field from data and `all` category (using the `one` relation on `EmojiRegistry`)